packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -459,109 +464,171 @@ export class SDKClient {
         }
       }
 
-      this.logger.trace(
-        `${requestIdPrefix} ${transaction.transactionId} ${callerName} ${transactionType} status: ${sdkClientError.status} (${sdkClientError.status._code}), cost: ${transactionFee}`,
+      // log main error
+      this.logger.debug(
+        `${requestId} Fail to execute ${transactionType} transaction: transactionId=${transactionId.flat()}, callerName=${callerName}, transactionType=${transactionType}, status=${
+          sdkClientError.status
+        }(${sdkClientError.status._code}), cost=${transactionFee} tinybars, gasUsed=${gasUsed}`,
       );
-      if (e instanceof JsonRpcError) {
-        throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
+
+      // Throw WRONG_NONCE error as more error handling logic for WRONG_NONCE is awaited in eth.sendRawTransactionErrorHandler(). Otherwise, move on and return transactionResponse eventually.
+      if (e.status && e.status.toString() === constants.TRANSACTION_RESULT_STATUS.WRONG_NONCE) {
+        throw sdkClientError;
+      } else {
+        if (!transactionResponse) {
+          this.logger.error(`${requestId} Transaction execution returns a null value for transaction ${transactionId}`);
+          throw predefined.INTERNAL_ERROR(
+            `${requestId} Transaction execution returns a null value for transaction ${transactionId}`,
+          );
+        }
+        return transactionResponse;
+      }
+    } finally {
+      /**
+       * @note Capturing the charged transaction fees at the end of the flow ensures these fees are eventually
+       *       captured in the metrics and rate limiter class, even if SDK transactions fail at any point.
+       */
+      if (transactionFee !== 0) {
+        this.logger.trace(
+          `${requestId} Capturing HBAR charged transaction fee: transactionId=${transactionId.flat()}, transactionType=${transactionType}, callerName=${callerName}, txChargedFee=${transactionFee} tinybars`,
+        );
+        this.hbarLimiter.addExpense(transactionFee, currentDateNow);
+        this.captureMetrics(
+          SDKClient.transactionMode,
+          transactionType,
+          Status.Success,
+          transactionFee,
+          gasUsed,
+          callerName,
+          interactingEntity,
+        );
       }
-      throw sdkClientError;
     }
   };
 
-  async executeGetTransactionRecord(
-    transactionResponse: TransactionResponse,
-    callerName: string,
-    interactingEntity: string,
-    txConstructorName: string,
-    requestId?: string,
-  ) {
-    const requestIdPrefix = formatRequestIdMessage(requestId);
+  async executeGetTransactionRecord(transactionResponse: TransactionResponse, callerName: string, requestId: string) {
     const currentDateNow = Date.now();
     let gasUsed: any = 0;
     let transactionFee: number = 0;
     const transactionId: string = transactionResponse.transactionId.toString();
 
+    const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.recordMode, callerName);
```

Comment:
Interesting, now i'm wondering why we have a limit in here at all or if we're handling the logic right. Will rate limiting on getRecord cause a valid transaction flow to fail with an error?
Example you Submit an `EthereumTransaction` successfully but hit a rate limit on getRecord. What happens?

Reply:
Ah yeah good point! hmm.. well to answer your question, if it hits a rate limit on `getRecord`, it will throw an `HBAR_RATE_LIMIT` error all the way to the top level and won't return any hashes, only the `jsonRpcError`. This is not ideal because the transaction was successfully submitted to the network. The `getRecord` function is like a secondary operation meant to retrieve the transaction fee for metrics purposes, and the charge is not too big.

Seems like this check is not necessary at this step.

Reply:
removed this check in the new update.